### PR TITLE
Scancore actions logging for the AI

### PR DESCRIPTION
**Pull request by Deezzir**
The changes to accommodate the action logging for AI training

### CODE REVIEW

**Comment by digimer** [comment]
Can you resolve conflicts? Then I'll merge.

**Comment by Deezzir** [comment]
Resolved

**Comment by digimer** [comment]
I'm worry, but can you rebase with main? It's not letting me merge automatically.

**Comment by Deezzir** [comment]
Can you try again? I’ve updated it

**Comment by digimer** [comment]
Thanks!

File: Anvil/Tools/Actions.pm [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+	}
+
+	if (not $peer_node_name)
+	{
+		$anvil->Log->entry({source => $THIS_FILE, line => __LINE__, level => 0, priority => "err", key => "log_0020", variables => { method => "Actions->insert_action_node_assume()", parameter => "peer_node_name" }});
+		return("!!error!!");
+	}
+
+	# Can we parse a node number from node_name or peer_node_name
+	my ($node_number) = $node_name =~ /(\d)/;
```
Comment (digimer) [inline Anvil/Tools/Actions.pm:123]: We don't require nodes to have specific names. We figure out which node is node 1 and 2 by comparing the host's host_uuid to the "anvil_node{1,2}_host_uuid" in the anvils table. If you call $anvil->Database->get_anvils() and then check if the host's host_uuid is in '$anvil->data->{anvils}{anvil_uuid}{$anvil_uuid}{anvil_node1_host_uuid}' or '$anvil->data->{anvils}{anvil_uuid}{$anvil_uuid}{anvil_node2_host_uuid}'.

File: Anvil/Tools/ScanCore.pm [hunk 2]
```diff
@@ -1654,6 +1666,13 @@ sub post_scan_analysis_node
 				$anvil->Alert->register({alert_level => "warning", message => "warning_0082", set_by => "ScanCore", variables => $variables});
 				$anvil->Email->send_alerts();
 				
+				if (!$action_log_disabled) {
```
Comment (digimer) [inline Anvil/Tools/ScanCore.pm:1669]: For formatting consistence and readability; We use: 

if (not $foo)
{
    ....
}

**Review by digimer — COMMENTED** [review]
Will be looking at the other commits next.

File: Anvil/Tools/Actions.pm [hunk 3]
```diff
@@ -121,7 +121,7 @@ sub insert_action_node_assume {
 
     # Can we parse a node number from node_name or peer_node_name
     my ($node_number) = $node_name =~ /(\d)/;
-    if (not $number or ($number ne "1" and $number ne "2")) {
+    if (not $node_number or ($node_number ne "1" and $node_number ne "2")) {
```
Comment (digimer) [inline Anvil/Tools/Actions.pm]: For consistency and code readability (again, heh); We wrap each conditional in it's own (). So here I'd use:

if ((not $node_number) or (($node_number ne "1") and ($node_number ne "2")))
{
   ...

**Review by digimer — CHANGES_REQUESTED** [review]
last comment.

File: Anvil/Tools/Actions.pm (hunk 3 above)
Comment (Deezzir) [reply Anvil/Tools/Actions.pm]: It's from the old commits, the latest version has that, I think

File: Anvil/Tools/ScanCore.pm (hunk 2 above)
Comment (Deezzir) [reply Anvil/Tools/ScanCore.pm]: Go that

**Review by digimer — DISMISSED** [review]

**Comment by fabbione** [comment]
retest this please

**Comment by digimer** [comment]
Two isses; Fisrt:
00:12 < fabbione> i see they merged main into their branch
00:12 < fabbione> that´s a no no
00:12 < fabbione> rebase
00:12 < fabbione> or die
00:12 < fabbione>  Merge branch 'main' into scancore-ai-actions 

Second, see compile errors here:

https://ci.kronosnet.org/job/anvil-build-functional-testing/4700/anvil-build-all-voting-functional-testing=anvil-bm/console

**Comment by fabbione** [comment]
retest this please

**Comment by digimer** [comment]
retest this please

**Pull request closed without merging**